test/Interpreter/escapable_generics_casting.swift

```diff
@@ -0,0 +1,50 @@
+// RUN: %target-run-simple-swift(-Xfrontend -sil-verify-all -enable-experimental-feature NoncopyableGenerics -enable-experimental-feature NonescapableTypes) | %FileCheck %s
+// RUN: %target-run-simple-swift(-O -Xfrontend -sil-verify-all -enable-experimental-feature NoncopyableGenerics -enable-experimental-feature NonescapableTypes) | %FileCheck %s
+
+// REQUIRES: executable_test, asserts
+
+protocol P {
+  func speak()
+}
+
+extension P {
+  func speak() { print("hello") }
+}
+
+struct Nonescapable: ~Escapable {}
+struct Ordinary {}
+
+struct Dog<T: ~Escapable>: Escapable {}
+extension Dog: P where T: Escapable {}
+
+func attemptCall(_ a: Any) {
+  if let value = a as? P {
+    value.speak()
+    return
+  }
+  print("failed to cast (attemptCall)")
+}
+
+defer { main({}) }
+func main(_ noEscapeFunc: () -> Void) {
+  // CHECK: hello
+  attemptCall(Dog<Ordinary>())
+
+  // CHECK: failed to cast (attemptCall)
+  attemptCall(Dog<Nonescapable>())
+
+  // CHECK: function types
+  print("function types")
```

Comment:
It's effectively a way to resync FileCheck with that place in the source file. This test should probably switch over to the unit test machinery instead of FileCheck.